### PR TITLE
Ability to set max number of active toasts

### PR DESCRIPTION
**Pull request by jamesgiu**
I've used BlueprintJS for work projects and this little idea has been mentioned a few times at work so I thought I'd see what you guys think. Also doubles as a Hacktoberfest contribution!

#### Checklist

- [x] Includes tests
- [x] Update documentation

#### Changes proposed in this pull request:

Ability to set a maximum number of active toasts being displayed at once. Any new toast effectively 'pushes out' the oldest one.

#### Reviewers should focus on:

Is 5 a suitable default limit? Or should default be no limit (as it is right now in BlueprintJS)?
Does it make sense for a user to specify '0' as an argument? I've covered for the use case, just in case

#### Screenshot
ezgif-1-12f9cee6fc3a

### CODE REVIEW

**Comment by palantirtech** [comment]
Thanks for your interest in [palantir/blueprint](https://github.com/palantir/blueprint), @jamesgiu! Before we can accept your pull request, you need to sign our contributor license agreement - just visit https://cla.palantir.com/ and follow the instructions. Once you sign, I'll automatically update this pull request.

File: packages/core/src/components/toast/toaster.tsx [hunk 1]
```diff
@@ -101,6 +109,7 @@ export class Toaster extends AbstractPureComponent2<IToasterProps, IToasterState
     public static defaultProps: IToasterProps = {
         autoFocus: false,
         canEscapeKeyClear: true,
+        maxToasts: 5,
```
Comment (adidahiya) [inline packages/core/src/components/toast/toaster.tsx]: this is technically a breaking change in behavior. I think you should not set a default, and if `maxToasts === undefined`, then do not enforce a limit.

File: packages/core/src/components/toast/toaster.tsx [hunk 2]
```diff
@@ -133,6 +142,12 @@ export class Toaster extends AbstractPureComponent2<IToasterProps, IToasterState
     private toastId = 0;
 
     public show(props: IToastProps, key?: string) {
+        // show no toasts if max toasts is less than 1.
+        if (this.props.maxToasts < 1) {
```
Comment (adidahiya) [inline packages/core/src/components/toast/toaster.tsx]: this validation should happen in `validateProps` (invoked via the super class `AbstractPureComponent2`). if `props.maxToasts < 1` I think you should fail loudly and throw a validation error (some other components do this).

**Comment by jamesgiu** [comment]
Thanks for your review @adidahiya!

**Comment by adidahiya** [comment]
thanks for the PR @jamesgiu!

**Pull request merged**